Client/SiteImageHelper.swift

```diff
@@ -43,19 +43,17 @@ class SiteImageHelper {
         switch imageType {
         case .heroImage:
             fetchHeroImage(for: site) { image, result in
-                guard let heroImage = image else { return }
-                didCompleteFetch = result ?? false
+                didCompleteFetch = result
```

Comment:
Is there a fallback mechanism as I feel returning nil is not the best option considering we use this at places where we expect to see a favicon. One way to deal with this is instead of returning nil we should return letter favicon.  

https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Frontend/Home/HistoryHighlights/FxHomeHistoryHighlightsCollectionCell.swift#L246-L248

let me know if I am missing anything, happy to understand more why returning nil image is better

Reply:
We already have a fallback within the call to getFaviconImage in SQLiteHistory https://github.com/mozilla-mobile/firefox-ios/blob/main/Storage/SQL/SQLiteHistoryFavicons.swift#L149-L172 
So in order we try to:
1. fetch from the bundled favicons
2. fetch from the database
3. scrape from the web page
4. if all fails then we generate a default favicon with the website letter

I agree we should have a refactor at one point to avoid return a `maybe` of deferred type and instead return an image all the time, since fallback is in place but can be easy to miss here.